scripts/ethEndToEndTest.js

```diff
@@ -0,0 +1,212 @@
+    const deposition = await peg.deposit(ethAddress, depositAmount, cennznetAddress, {
+        gasLimit: 500000,
+        value: depositAmount
+    });
+    const depositTxHash = deposition.hash;
+    console.log('depositTxHash:::',depositTxHash);
+
+    const ethBalanceAfter = await ethers.provider.getBalance(peg.address);
+    console.log('Balance after::', ethBalanceAfter.toString());
+
+
+
+    /***************** CENNZnet Setup **********************/
+        // connecting to local chain
+    const api = await Api.create();
+    const keyring = new Keyring({type: 'sr25519'});
+    const alice = keyring.addFromUri('//Alice');
+
+    /********************* Need the following set up *****************************/
+    /********************* only for local chain **********************************/
+    /********************* wont need this for rata/nikau/azalea ******************/
+    const transaction1 = api.tx.erc20Peg.activateDeposits(true);
+    const transaction2 = api.tx.erc20Peg.activateWithdrawals(true);
+    const transaction3 = api.tx.erc20Peg.setContractAddress(peg.address);
+    const transaction4 = api.tx.ethBridge.setEventConfirmations(0); // Hardhat only makes blocks when txs are sent
+    const batchBridgeActivationEx = api.tx.utility.batch([
+        transaction1,
+        transaction2,
+        transaction3,
+        transaction4
+    ]);
+
+    await new Promise(async (resolve, reject) => {
+        await api.tx.sudo.sudo(batchBridgeActivationEx).signAndSend(alice, async ({status, events}) => {
+            if (status.isInBlock) {
+                events.forEach(({phase, event: {data, method, section}}) => {
+                    console.log('\t', phase.toString(), `: ${section}.${method}`, data.toString());
+                    resolve();
+                });
+            }
+        });
+    });
+    /********************* Set up complete *******************************************/
+
+    /********** new Token Id that would be minted on deposition at CENNZnet **********/
+    const testTokenId = await api.query.genericAsset.nextAssetId();
+    const claim = {
+        tokenAddress: ethAddress,
+        amount: depositAmount,
+        beneficiary: cennznetAddress
+    };
+
+    console.log('testTokenId::', testTokenId.toString());
+
+    /***************** Make Deposit on CENNZnet **********************/
+    let eventClaimId;
+    let nonce = await api.rpc.system.accountNextIndex(alice.address);
+    await new Promise(async (resolve, reject) => {
+        await api.tx.erc20Peg.depositClaim(depositTxHash, claim).signAndSend(alice, {nonce}, async ({status, events}) => {
+            if (status.isInBlock) {
+                for (const {event: {method, section, data}} of events) {
+                    console.log('\t', `: ${section}.${method}`, data.toString());
+                    if (section === 'erc20Peg' && method == 'Erc20Claim') {
+                        eventClaimId = data[0];
```

Comment:
is it possible to checker the claimer/sender address here also?

Reply:
done